gentoopm/bash/bashserver.py

```diff
@@ -59,7 +59,11 @@ def _read1(self):
 		f = self._bashproc.stdout
 		buf = b' '
 		while not buf.endswith(b'\0'):
-			buf += f.read(1)
+			x = f.read(1)
+			if len(x) < 1:
+				# end-of-file
+				break
```

Comment:
I think you should raise the whatever bash exception we have in here. Passing partial data sounds like a bad idea.